sklearn/linear_model/tests/test_ransac.py

```diff
@@ -153,9 +153,14 @@ def test_ransac_resid_thresh_no_inliers():
     ransac_estimator = RANSACRegressor(base_estimator, min_samples=2,
                                        residual_threshold=0.0, random_state=0)
 
-    assert_raises_regexp(ValueError,
-                    "No inliers.*residual_threshold.*0\.0",
-                    ransac_estimator.fit, X, y)
+    msg = (
+        "RANSAC could not find valid consensus set, because"
+        " either the `residual_threshold` rejected all the samples or"
+        " `is_data_valid` and `is_model_valid` returned False for all"
```

Comment:
Oh, sorry, my fault. The current message is ok, though your suggestion is even better!